### PR TITLE
Crud competitions#4

### PR DESCRIPTION
**Pull request by Serban681**

### CODE REVIEW

File: src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private lateinit var competitionService: CompetitionService
+
+    @Autowired
+    private lateinit var competitionMapper: CompetitionMapper
+
+    @GetMapping
+    fun getAll(): List<CompetitionDto> = competitionService.getAll()
+
+    @GetMapping("/{id}")
+    fun getOneById(@PathVariable id: Long): CompetitionDto {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt:34]: You can make this method one line of you get rid of the intermediary variable. Something like this:

```kotlin
competitionMapper.mapCompetitionToCompetitionDto(competitionService.getOne(id))
```

File: src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        val competition = competitionService.getOne(id)
+
+        return competitionMapper.mapCompetitionToCompetitionDto(competition)
+    }
+
+    @PostMapping
+    fun create(@RequestBody dto: CompetitionRequest): ResponseEntity<CompetitionDto> {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt:41]: Make one line

File: src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        val competition = competitionService.create(dto)
+
+        return ResponseEntity(competitionMapper.mapCompetitionToCompetitionDto(competition), HttpStatus.OK)
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt:44]: For post return `HttpStatus.CREATED` on success

File: src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @PutMapping("/{id}")
+    fun update(@PathVariable id: Long, @RequestBody dto: CompetitionRequest): ResponseEntity<CompetitionDto> {
+        val newCompetition = competitionService.update(id, dto)
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/controller/CompetitionController.kt:49]: Make one line

File: src/main/kotlin/ro/hibyte/betting/dto/EventDTO.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package ro.hibyte.betting.dto
+
+import ro.hibyte.betting.entity.Status
+import java.time.Instant
+
+data class EventDTO(
+    var eventId: Long? =null,
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/dto/EventDTO.kt:7]: Do a `Ctrl+Alt+L` here to automatically format file

File: src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import java.time.Instant
+
+@Service
+class CompetitionMapper {
+    @Autowired
+    private lateinit var eventRepository: EventRepository
+    @Autowired
+    private lateinit var userProfileRepository: UserProfileRepository
+
+    fun getEventsFromIds(eventIDs: List<Long>): List<Event> = eventIDs.map { id -> eventRepository.findById(id).orElseThrow{ EntityNotFoundException("Event", id) } }.toList()
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt:24]: Split line like this:

```kotlin
fun getEventsFromIds(eventIDs: List<Long>): List<Event> = 
    eventIDs.map { 
        id -> eventRepository.findById(id).orElseThrow{ EntityNotFoundException("Event", id) } 
    }.toList()

File: src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    fun getUserProfilesFromIds(userIDs: List<Long>): Set<UserProfile> = userIDs.map { id -> userProfileRepository.findById(id).orElseThrow { EntityNotFoundException("User Profile", id) } }.toSet()
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt:25]: Same here

File: src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    fun mapCompetitionRequestToCompetition(competitionRequest: CompetitionRequest): Competition {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt:27]: You can use one line function syntax

File: src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            userGroups = competitionRequest.userGroups,
+            userProfiles = competitionRequest.userProfiles,
+            events = getEventsFromIds(competitionRequest.events),
+            created = Timestamp(System.currentTimeMillis()),
+            lastModified = Timestamp(System.currentTimeMillis()),
+            status = competitionRequest.status,
+        )
+    }
+
+    fun mapCompetitionRequestToCompetition(competitionRequest: CompetitionRequest, initialCompetition: Competition): Competition {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt:42]: One line method syntax

File: src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            users = getUserProfilesFromIds(competitionRequest.users),
+            userGroups = competitionRequest.userGroups,
+            userProfiles = competitionRequest.userProfiles,
+            events = getEventsFromIds(competitionRequest.events),
+            created = initialCompetition.created,
+            lastModified = Timestamp(System.currentTimeMillis()),
+            status = competitionRequest.status,
+        )
+    }
+    fun mapCompetitionToCompetitionDto(competition: Competition): CompetitionDto {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/mapper/CompetitionMapper.kt:57]: One line method syntax

File: src/main/kotlin/ro/hibyte/betting/service/BetService.kt [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+import ro.hibyte.betting.repository.UserProfileRepository
+
+@Service
+class BetService(private val betRepository: BetRepository,
+                 private val betTypeRepository: BetTypeRepository
+) {
+
+    fun getAll(): List<Bet> = betRepository.findAll()
+
+    fun get(betId: Long): Bet {
```
Comment (DrBaxR) [inline src/main/kotlin/ro/hibyte/betting/service/BetService.kt:20]: One line syntax

File: src/main/resources/application.yml [hunk 12]
```diff
@@ -2,7 +2,7 @@ spring:
   datasource:
     url: jdbc:postgresql://localhost:5432/betting
     username: postgres
-    password: admin
+    password: 1234
```
Comment (DrBaxR) [inline src/main/resources/application.yml:5]: Don't commit local database/any password, please change to what it was before (let me know when you get to this, I can show you a trick that helps with this)

**Review by DrBaxR — CHANGES_REQUESTED** [review]
Ignore the comments I left on files that are not yours/that don't apply to your files

**Comment by DrBaxR** [comment]
Did you close PR #12, made the changes requested there and opened a new PR? If so, you don't need to do that, if you make the changes on the same branch and push on it, the PR will be automatically updated

**Comment by Serban681** [comment]
I had problems when I was trying to pull code made by others. So I decided to create another branch and a new PR.

**Review by DrBaxR — APPROVED** [review]

**Review by DanceaRadu — APPROVED** [review]
The creator of the Competition is not set properly, but I assume this will be changed later, once the UserProfile tasks are completed. Other than that, it seems fine.

**Pull request merged**